### ISSUE TITLE
Joining a room should be more transparent for the user

### ISSUE THREAD

**Issue by devnoname120** [vector-im/riot-web#3484]
### Description

When I access a room, there is a banner explaining that I need to join the room in order to send messages.

Instead, I would like this banner not to appear, and automatically join the room as I type my first message.

Same for private discussions, I'm not sure why I need to manually click on join the conversation, this feels annoying, especially as a former IRC client user.

**Comment by lampholder** [comment]
I believe the intention of this behaviour is to give the user an opportunity to 'peek' into the room before committing themselves to visibly joining.

The act of joining is therefore then a very deliberate action, so it _should_ be clear to the user that they are making a choice to change their state, from peeking to a full member of the room.

@ara4n - can you confirm or deny my interpretation of this feature?

**Comment by ara4n** [comment]
correct. and if peeking is not enabled, the prompt serves to let users explicitly opt into invitations or following room urls etc.

however, if the room has no peeking *and* the user explicitly tried to join it from the UI (by /join, or using the room directory, or following a link) i think it would be fine to join without the confirmation screen.

**Comment by devnoname120** [comment]
I find it the most annoying for PM's. For example if a IRC user sends me a message, I find it a burden - especially for one-time one-line chats - to click on 'join this room' and needing to wait for some short time before the messages appear.

I think that the default desired behavior for clicking on join from the UI or typing `/join` is actually *joining* the room, not *peeking* in it. Maybe a `/peek` command could be added?

**Comment by lampholder** [comment]
I think we could probably all agree that it would be good for the behaviour of this feature to be configurable according to individual user preferences.

The explicit opt-in is a safe default (having to accept requests to send inbound messages limits risk of abuse), but I'm sure some power users would prefer to be dropped straight into the chat, so let's wrap this up by defining it as a 'configurable peek/join behaviour' feature request and adding to the priority list :)

I'm assigning this P3 to represent that we're not planning on doing it in the immediate term, but would be happy to see pull requests from the community :)
